src/poolState.ts

```diff
@@ -52,6 +59,22 @@ export function parseGlobalPoolState(rawState: any[]): AppInternalState {
   const CONFIG = state.CONFIG ?? "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAe";
   delete state.CONFIG;
 
+  if (state.CONTRACT_NAME) {
+    state.CONTRACT_NAME = decode(Buffer.from(state.CONTRACT_NAME, "base64"));
+  }
+
+  if (state.ADMIN) {
+    state.ADMIN = algosdk.encodeAddress(
+      Buffer.from(Buffer.from(state.ADMIN, "base64")),
+    );
```

Comment:
How about making helper function for this? While adding admin actions guys had problems at first with address encoding - this might be helpful in the future

Reply:
done